bool left
bool right
bool mid